From b62048701cea5024383e19314592f2edcd9810fd Mon Sep 17 00:00:00 2001
From: Stephan Bergmann <sbergman@redhat.com>
Date: Sat, 16 Feb 2013 22:39:13 +0100
Subject: [PATCH] Work around problem with boost::shared_array(NULL) ctor and
 Boost 1.53.0

...claiming

  template<class Y> boost::shared_array<T>::shared_array(Y*)

is not a viable option due to mismatched types 'Y*' and 'long int'

Change-Id: I8db321cd25cd73c84fa2a3124c9ec1018c131d5f
---
 vcl/source/gdi/svgdata.cxx | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/vcl/source/gdi/svgdata.cxx b/vcl/source/gdi/svgdata.cxx
index 01707e6..e64646f 100644
--- a/vcl/source/gdi/svgdata.cxx
+++ b/vcl/source/gdi/svgdata.cxx
@@ -166,7 +166,7 @@ SvgData::SvgData(const SvgDataArray& rSvgDataArray, sal_uInt32 nSvgDataArrayLeng
 
 //////////////////////////////////////////////////////////////////////////////
 SvgData::SvgData(const OUString& rPath):
-    maSvgDataArray(NULL),
+    maSvgDataArray(),
     mnSvgDataArrayLength(0),
     maPath(rPath),
     maRange(),
-- 
1.8.1.2


From d4bab97023e3569571a92551040574b20aceca7c Mon Sep 17 00:00:00 2001
From: Mark Wright <markwright@internode.on.net>
Date: Sun, 10 Feb 2013 02:04:02 +1100
Subject: [PATCH] fix compile for change to boost 1.53.0 declaring smart
 pointer operator bool as explicity for C++11 compilers

Change-Id: If2c3ad68b2ffea645a9f2035cd802553edc0ee79
Reviewed-on: https://gerrit.libreoffice.org/2064
Tested-by: LibreOffice gerrit bot <gerrit@libreoffice.org>
Reviewed-by: Norbert Thiebaud <nthiebaud@gmail.com>
---
 comphelper/inc/comphelper/scoped_disposing_ptr.hxx | 2 +-
 sc/source/filter/excel/xechart.cxx                 | 2 +-
 sc/source/filter/excel/xehelper.cxx                | 2 +-
 sc/source/filter/excel/xichart.cxx                 | 2 +-
 sc/source/filter/inc/xichart.hxx                   | 4 ++--
 slideshow/source/engine/animatedsprite.cxx         | 2 +-
 slideshow/source/engine/shapes/viewshape.cxx       | 2 +-
 slideshow/source/engine/shapesubset.cxx            | 2 +-
 slideshow/source/engine/slide/slideanimations.cxx  | 2 +-
 slideshow/source/inc/shapeattributelayer.hxx       | 2 +-
 slideshow/source/inc/shapeattributelayerholder.hxx | 2 +-
 sw/source/core/inc/bookmrk.hxx                     | 2 +-
 12 files changed, 13 insertions(+), 13 deletions(-)

diff --git a/comphelper/inc/comphelper/scoped_disposing_ptr.hxx b/comphelper/inc/comphelper/scoped_disposing_ptr.hxx
index 9b4fe1b..6c34074 100644
--- a/comphelper/inc/comphelper/scoped_disposing_ptr.hxx
+++ b/comphelper/inc/comphelper/scoped_disposing_ptr.hxx
@@ -78,7 +78,7 @@ public:
 
     operator bool () const
     {
-        return m_aItem;
+        return static_cast< bool >(m_aItem);
     }
 
     virtual ~scoped_disposing_ptr()
diff --git a/sc/source/filter/excel/xechart.cxx b/sc/source/filter/excel/xechart.cxx
index fd99e8b..75a2eb2 100644
--- a/sc/source/filter/excel/xechart.cxx
+++ b/sc/source/filter/excel/xechart.cxx
@@ -683,7 +683,7 @@ void XclExpChEscherFormat::Convert( const ScfPropertySet& rPropSet, XclChObjectT
 
 bool XclExpChEscherFormat::IsValid() const
 {
-    return maData.mxEscherSet;
+    return static_cast< bool >(maData.mxEscherSet);
 }
 
 void XclExpChEscherFormat::Save( XclExpStream& rStrm )
diff --git a/sc/source/filter/excel/xehelper.cxx b/sc/source/filter/excel/xehelper.cxx
index c493560..81e9ac2 100644
--- a/sc/source/filter/excel/xehelper.cxx
+++ b/sc/source/filter/excel/xehelper.cxx
@@ -303,7 +303,7 @@ rtl::OUString XclExpHyperlinkHelper::ProcessUrlField( const SvxURLField& rUrlFie
     if( GetBiff() == EXC_BIFF8 )    // no HLINK records in BIFF2-BIFF7
     {
         // there was/is already a HLINK record
-        mbMultipleUrls = mxLinkRec;
+        mbMultipleUrls = static_cast< bool >(mxLinkRec);
 
         mxLinkRec.reset( new XclExpHyperlink( GetRoot(), rUrlField, maScPos ) );
 
diff --git a/sc/source/filter/excel/xichart.cxx b/sc/source/filter/excel/xichart.cxx
index 63c9c43..f2321bf 100644
--- a/sc/source/filter/excel/xichart.cxx
+++ b/sc/source/filter/excel/xichart.cxx
@@ -2690,7 +2690,7 @@ void XclImpChTypeGroup::Finalize()
     maType.Finalize( bStockChart );
 
     // extended type info
-    maTypeInfo.Set( maType.GetTypeInfo(), mxChart3d, false );
+    maTypeInfo.Set( maType.GetTypeInfo(), static_cast< bool >(mxChart3d), false );
 
     // reverse series order for some unstacked 2D chart types
     if( maTypeInfo.mbReverseSeries && !Is3dChart() && !maType.IsStacked() && !maType.IsPercent() )
diff --git a/sc/source/filter/inc/xichart.hxx b/sc/source/filter/inc/xichart.hxx
index 57538f0..7c44412 100644
--- a/sc/source/filter/inc/xichart.hxx
+++ b/sc/source/filter/inc/xichart.hxx
@@ -1230,9 +1230,9 @@ public:
     /** Returns true, if the axis contains caption labels. */
     inline bool         HasLabels() const { return !mxTick || mxTick->HasLabels(); }
     /** Returns true, if the axis shows its major grid lines. */
-    inline bool         HasMajorGrid() const { return mxMajorGrid; }
+    inline bool         HasMajorGrid() const { return static_cast< bool >(mxMajorGrid); }
     /** Returns true, if the axis shows its minor grid lines. */
-    inline bool         HasMinorGrid() const { return mxMinorGrid; }
+    inline bool         HasMinorGrid() const { return static_cast< bool >(mxMinorGrid); }
 
     /** Creates an API axis object. */
     XAxisRef            CreateAxis( const XclImpChTypeGroup& rTypeGroup, const XclImpChAxis* pCrossingAxis ) const;
diff --git a/slideshow/source/engine/animatedsprite.cxx b/slideshow/source/engine/animatedsprite.cxx
index e63d600..5d421c2 100644
--- a/slideshow/source/engine/animatedsprite.cxx
+++ b/slideshow/source/engine/animatedsprite.cxx
@@ -151,7 +151,7 @@ namespace slideshow
                 }
             }
 
-            return mpSprite;
+            return static_cast< bool >(mpSprite);
         }
 
         void AnimatedSprite::setPixelOffset( const ::basegfx::B2DSize& rPixelOffset )
diff --git a/slideshow/source/engine/shapes/viewshape.cxx b/slideshow/source/engine/shapes/viewshape.cxx
index 20c73de..621535f 100644
--- a/slideshow/source/engine/shapes/viewshape.cxx
+++ b/slideshow/source/engine/shapes/viewshape.cxx
@@ -178,7 +178,7 @@ namespace slideshow
                 }
             }
 
-            return io_rCacheEntry.mpRenderer;
+            return static_cast< bool >(io_rCacheEntry.mpRenderer);
         }
 
         bool ViewShape::draw( const ::cppcanvas::CanvasSharedPtr&   rDestinationCanvas,
diff --git a/slideshow/source/engine/shapesubset.cxx b/slideshow/source/engine/shapesubset.cxx
index c5636cc..da28159 100644
--- a/slideshow/source/engine/shapesubset.cxx
+++ b/slideshow/source/engine/shapesubset.cxx
@@ -104,7 +104,7 @@ namespace slideshow
                     maTreeNode );
             }
 
-            return mpSubsetShape;
+            return static_cast< bool >(mpSubsetShape);
         }
 
         void ShapeSubset::disableSubsetShape()
diff --git a/slideshow/source/engine/slide/slideanimations.cxx b/slideshow/source/engine/slide/slideanimations.cxx
index 5f5e9f5..7d4c788 100644
--- a/slideshow/source/engine/slide/slideanimations.cxx
+++ b/slideshow/source/engine/slide/slideanimations.cxx
@@ -74,7 +74,7 @@ namespace slideshow
 
             SHOW_NODE_TREE( mpRootNode );
 
-            return mpRootNode;
+            return static_cast< bool >(mpRootNode);
         }
 
         bool SlideAnimations::isAnimated() const
diff --git a/slideshow/source/inc/shapeattributelayer.hxx b/slideshow/source/inc/shapeattributelayer.hxx
index 88405f8..d725255 100644
--- a/slideshow/source/inc/shapeattributelayer.hxx
+++ b/slideshow/source/inc/shapeattributelayer.hxx
@@ -467,7 +467,7 @@ namespace slideshow
             // ShapeAttributeLayer(const ShapeAttributeLayer&);
             // ShapeAttributeLayer& operator=( const ShapeAttributeLayer& );
 
-            bool haveChild() const { return mpChild; }
+            bool haveChild() const { return static_cast< bool >(mpChild); }
             void updateStateIds();
 
             template< typename T > T calcValue( const T&                   rCurrValue,
diff --git a/slideshow/source/inc/shapeattributelayerholder.hxx b/slideshow/source/inc/shapeattributelayerholder.hxx
index e53be46..22ce4f3 100644
--- a/slideshow/source/inc/shapeattributelayerholder.hxx
+++ b/slideshow/source/inc/shapeattributelayerholder.hxx
@@ -83,7 +83,7 @@ namespace slideshow
                 if( mpShape )
                     mpAttributeLayer = mpShape->createAttributeLayer();
 
-                return mpAttributeLayer;
+                return static_cast< bool >(mpAttributeLayer);
             }
 
             ShapeAttributeLayerSharedPtr get() const
diff --git a/sw/source/core/inc/bookmrk.hxx b/sw/source/core/inc/bookmrk.hxx
index 6937a19..bab92bb 100644
--- a/sw/source/core/inc/bookmrk.hxx
+++ b/sw/source/core/inc/bookmrk.hxx
@@ -75,7 +75,7 @@ namespace sw {
 
             virtual bool IsCoveringPosition(const SwPosition& rPos) const;
             virtual bool IsExpanded() const
-                { return m_pPos2; }
+                { return static_cast< bool >(m_pPos2); }
 
             virtual void SetName(const ::rtl::OUString& rName)
                 { m_aName = rName; }
-- 
1.8.1.2


From faf72664d11deaa570a0f8c7e702ff07bf5914c8 Mon Sep 17 00:00:00 2001
From: Stephan Bergmann <sbergman@redhat.com>
Date: Sun, 17 Feb 2013 10:49:34 +0000
Subject: Missing include in liborcus when building against Boost 1.53.0

Change-Id: Ie6c944d53d6e2fcdd161f674e32c9fbfd761d844
---
diff --git a/liborcus/UnpackedTarball_orcus.mk b/liborcus/UnpackedTarball_orcus.mk
index ca5232d..de0788a 100644
--- a/liborcus/UnpackedTarball_orcus.mk
+++ b/liborcus/UnpackedTarball_orcus.mk
@@ -18,6 +18,10 @@ orcus_patches :=
 orcus_patches += liborcus_0.1.0-configure.patch
 # disable boost "auto lib" in MSVC build
 orcus_patches += liborcus_0.1.0-boost_disable_auto_lib.patch
+# <https://gitorious.org/orcus/orcus/merge_requests/2#
+# f60d6eecee72349993a392a9a63ddf3383d3b8c8-
+# f60d6eecee72349993a392a9a63ddf3383d3b8c8@2>:
+orcus_patches += liborcus_0.1.0-boost-include.patch.1
 
 $(eval $(call gb_UnpackedTarball_fix_end_of_line,orcus,\
 	vsprojects/liborcus-static-nozip/liborcus-static-nozip.vcproj \
diff --git a/liborcus/liborcus_0.1.0-boost-include.patch.1 b/liborcus/liborcus_0.1.0-boost-include.patch.1
new file mode 100644
index 0000000..07558b0
--- a/dev/null
+++ b/liborcus/liborcus_0.1.0-boost-include.patch.1
@@ -0,0 +1,26 @@
+From f60d6eecee72349993a392a9a63ddf3383d3b8c8 Mon Sep 17 00:00:00 2001
+From: Stephan Bergmann <sbergman@redhat.com>
+Date: Sat, 16 Feb 2013 13:25:44 +0100
+Subject: [PATCH] Missing include <boost/noncopyable.hpp>
+
+(at least when compiling against Boost 1.53.0)
+---
+ src/liborcus/dom_tree.cpp | 2 ++
+ 1 file changed, 2 insertions(+)
+
+diff --git a/src/liborcus/dom_tree.cpp b/src/liborcus/dom_tree.cpp
+index ecf7beb..14725ad 100644
+--- a/src/liborcus/dom_tree.cpp
++++ b/src/liborcus/dom_tree.cpp
+@@ -34,6 +34,8 @@
+ #include <iostream>
+ #include <sstream>
+ 
++#include <boost/noncopyable.hpp>
++
+ using namespace std;
+ 
+ namespace orcus {
+-- 
+1.8.1.2
+
--
cgit v0.9.0.2-2-gbebe

